pkg/controllermanager/controller/shoot/shoot_control.go

```diff
@@ -200,160 +153,142 @@ func (c *Controller) updateShootStatusProcessing(shoot *gardenv1beta1.Shoot, mes
-	return &defaultControl{k8sGardenClient, k8sGardenInformers, secrets, imageVector, identity, config, gardenerNamespace, recorder}
-}
-
-type defaultControl struct {
-	k8sGardenClient    kubernetes.Interface
-	k8sGardenInformers gardeninformers.Interface
-	secrets            map[string]*corev1.Secret
-	imageVector        imagevector.ImageVector
-	identity           *gardenv1beta1.Gardener
-	config             *config.ControllerManagerConfiguration
-	gardenerNamespace  string
-	recorder           record.EventRecorder
-}
-
-func (c *defaultControl) ReconcileShoot(shootObj *gardenv1beta1.Shoot, key string, operation *operation.Operation) (bool, error) {
-	key, err := cache.MetaNamespaceKeyFunc(shootObj)
-	if err != nil {
-		return true, err
-	}
-	operationID, err := utils.GenerateRandomString(8)
-	if err != nil {
-		return true, err
+	c.recorder.Event(shoot, corev1.EventTypeNormal, gardenv1beta1.EventDeleted, "Deleted Shoot cluster")
+	if err := c.updateShootStatusDeleteSuccess(o); err != nil {
+		o.Logger.WithError(err).Errorf("Could not update the Shoot status after deletion success")
+		return reconcile.Result{}, err
 	}
+	return reconcile.Result{}, nil
+}
 
+func (c *Controller) reconcileShoot(shoot *gardenv1beta1.Shoot, o *operation.Operation) (reconcile.Result, error) {
 	var (
-		shoot         = shootObj.DeepCopy()
-		shootLogger   = logger.NewShootLogger(logger.Logger, shoot.Name, shoot.Namespace, operationID)
-		operationType = gardencorev1alpha1helper.ComputeOperationType(shoot.ObjectMeta, shoot.Status.LastOperation)
+		operationType       = gardencorev1alpha1helper.ComputeOperationType(shoot.ObjectMeta, shoot.Status.LastOperation)
+		failedOrIgnored     = common.IsShootFailed(shoot) || common.ShouldIgnoreShoot(c.respectSyncPeriodOverwrite(), shoot)
+		reconcileNotAllowed = c.reconcileInMaintenanceOnly() && (!common.IsOperationOngoing(shoot) && !common.IsNowInEffectiveShootMaintenanceTimeWindow(shoot))
+		allowedToUpdate     = !failedOrIgnored && !reconcileNotAllowed
 	)
 
-	logger.Logger.Infof("[SHOOT RECONCILE] %s", key)
-	shootJSON, _ := json.Marshal(shoot)
-	shootLogger.Debugf(string(shootJSON))
-
-	// We check whether the Shoot's last operation status field indicates that the last operation failed (i.e. the operation
-	// will not be retried unless the shoot generation changes).
-	if shootIsFailed(shoot) {
-		if shoot.Status.Gardener.Version == version.Get().GitVersion {
-			shootLogger.Infof("Will not reconcile as the last operation has been set to '%s' and the generation has not changed since then.", gardencorev1alpha1.LastOperationStateFailed)
-			return false, nil
+	if err := c.checkSeedAndSyncClusterResource(shoot, o); err != nil {
+		o.Logger.WithError(err).Error("Could not check and sync Shoot with Seed")
```

Comment:
```suggestion
		o.Logger.WithError(err).Error("Could not prepare for shoot reconciliation")
```

Reply:
Also, it's not needed to log here again as you are returning the error anyways which will lead to a log message from the calling functions. As suggested above, let's change the message to make it better understandable from outside.

(same for `deleteShoot` function, I guess)